core/src/Cuda/Kokkos_Cuda_Instance.cpp

```diff
@@ -253,15 +279,18 @@ CudaInternalDevices::CudaInternalDevices() {
   // See 'cudaSetDeviceFlags' for host-device thread interaction
   // Section 4.4.2.6 of the CUDA Toolkit Reference Manual
 
-  KOKKOS_IMPL_CUDA_SAFE_CALL(cudaGetDeviceCount(&m_cudaDevCount));
+  CudaInternal::singleton().cuda_api_interface_safe_call<false, int *>(
+      &cudaGetDeviceCount, &m_cudaDevCount);
```

Comment:
Here and the one below are the only  calls that need `setCudaDevice=false` since `CudaInternal::m_CudaDev` has not been set.